### ISSUE TITLE
add varStatus to h:dataTable

### ISSUE THREAD

**Issue by glassfishrobot** [javaee/javaserverfaces-spec#62]
ranjith.mandala@sun.com 2004-12-23 23:30:33 GMT

RM> A DESCRIPTION OF THE REQUEST :
RM>
RM> There seems to be nothing in the h:dataTable tag that allows me easy
RM> access to the count of the current loop status.
RM>
RM> I used JSTL before JSF and JSTL provided a "varStatus" attribute
RM> which is so handy.
RM>
RM> JUSTIFICATION :
RM>
RM> Currently to overcome this I have to insert the numbering into each
RM> object to show the position of the object in the list. It would be
RM> much nicer if something like JSTL's forEach "varStatus" attribute
RM> was available to h:dataDTable.

This would mean creating something like LoopTagStatus [1] and exposing
an instance of it for each time through the UIData iteration.

We might be able to even use the actual LoopTagStatus interface because
we're now depending on J2EE 5.0.

[1]
[http://java.sun.com/products/jsp/jstl/1.1/docs/api/javax/servlet/jsp/jstl/core/LoopTagStatus.html](http://java.sun.com/products/jsp/jstl/1.1/docs/api/javax/servlet/jsp/jstl/core/LoopTagStatus.html)
#### Environment
Operating System: All
Platform: All
#### Affected Versions
[1.1]

**Comment by glassfishrobot** [comment]
Reported by @edburns

**Comment by glassfishrobot** [comment]
Was assigned to javaserverfowner

**Comment by glassfishrobot** [comment]
@edburns said:
RM> The original incident is at
RM> [http://webbugs.sfbay.sun.com/rt/incidentDisplay?incidentID=311624](http://webbugs.sfbay.sun.com/rt/incidentDisplay?incidentID=311624)

RM> (company - Salestech , email - jonlee33@yahoo.com)

craig.mcclanahan@sun.com 2004-12-24 01:00:14 GMT

CM> There is a mechanism by which you can get the current row index,
CM> involving the following strategy:

CM> (1) Use a "binding" attribute on the <h:dataTable>
CM> component, binding it to a property in a backing bean:
CM>
CM> <h:dataTable ... binding="#

{backing.table}"/>
CM>
CM> (2) In a backing bean stored under managed bean name "backing"
CM> create the corresponding property:
CM>
CM> private UIData table = null;
CM> public UIData getTable() { return this.table; }
CM> public void setTable(UIData table) { this.table = table; }
CM>
CM> (3) To create, for example, an output column that is the
CM> row index, use something like this:
CM>
CM> <h:dataTable ... binding="#{backing.table}

" ...>
CM> ...
CM> <h:column>
CM> <h:outputText value="#

{backing.table.rowIndex}

"/>
CM> </h:column>
CM> ...
CM> </h:dataTable>
CM>
CM> This works because the "rowIndex" property of the UIData component
CM> **is** the current row index (zero relative). Binding the component
CM> to a backing bean property makes it accessible to an EL expression
CM> that can retrieve this value.
CM>

**Comment by glassfishrobot** [comment]
@edburns said:
Even with Craig's suggestion. I still think there is value in investigating the
merits of this feature request. I'll work it into the schedule.

**Comment by glassfishrobot** [comment]
@edburns said:
downgrade to p4.

**Comment by glassfishrobot** [comment]
@edburns said:
Since JSTL ForEach now works properly with JSF, I am marking this as WONTFIX.

**Comment by glassfishrobot** [comment]
@edburns said:
See
<[https://javaserverfaces-spec-public.dev.java.net/issues/show_bug.cgi?id=521](https://javaserverfaces-spec-public.dev.java.net/issues/show_bug.cgi?id=521)>
for a fix for this, but on ui:repeat.

**Comment by glassfishrobot** [comment]
@edburns said:
Prepare to delete api subcomponent

**Comment by glassfishrobot** [comment]
@edburns said:
Move all to 1.2

**Comment by glassfishrobot** [comment]
Marked as **won't fix** on Thursday, March 4th 2010, 6:09:24 am

**Comment by glassfishrobot** [comment]
@manfredriem said:
Closing resolved issue out

**Comment by glassfishrobot** [comment]
This issue was imported from java.net JIRA JAVASERVERFACES_SPEC_PUBLIC-62
